docs/guides/interfaces.rst

```diff
@@ -1,10 +1,89 @@
-.. _interfaces_guide:
+.. _interfaces:
+========================
+Interfaces
+========================
 
-===================
-Interfaces guide
-===================
 
-This guide is currently a work in progress and only contains bullet points for now.
+One of the biggest strengths of Orquestra is its modularity. Integrating new backends, optimizers, ansatzes, etc. does not require changing the core code - it requires only creating a new module that conforms to existing interfaces and therefore can be used across the whole platform. Using our interfaces help you add whatever building blocks you need for your project and ensure that they will work well with other existing elements.
 
-Cost Functions in VQAs
-======================
\ No newline at end of file
+What is an interface?
+=====================
+
+Before we get into the details, let's explain what we mean by "interfaces" in this context. We use this word as a shorthand for "a way to implement certain concepts in such a way, that as long as two implementations conform to a particular interface, they can be swapped and the program should still work correctly". Obviously, various implementations might work differently - e.g.: one simulator might be faster than the other - but they should be undistinguishable when it comes to inputs, outputs and how they are being used.
+
+
+
+Benefits of interfaces
+======================
+
+Interface offers the ability to define behaviour that can be implemented by unrelated classes without forcing them to share a common class hierarchy. This results in code flexibility and faster development with easier testing options. Other benefits include:
+
+* Interoperability
+* Modularity
+* Dependencies decoupling
+
+
+Interface implementation
+========================
+
+Currently interfaces in Orquestra are defined in the following 3 main libraries:
+
+* `orquestra-quantum <https://github.com/zapatacomputing/orquestra-quantum>`_
+* `orquestra-opt <https://github.com/zapatacomputing/orquestra-opt>`_
+* `orquestra-vqa <https://github.com/zapatacomputing/orquestra-vqa>`_
+
+The interfaces are located in the ``api`` directory within each packages. They are implemented either as an ``abstract class`` or ``protocols``. `Abstract classes <https://docs.python.org/3/library/abc.html>`_ are classes that contain one or more abstract method. However, they do not contain any implementation. Subclasses are required to provide implementations for abstract methods. Abstract classes can also be considered blueprint classes. 
```

Comment:
I'm going to hell for pointing this out, but... Abstract class CAN contain an implementation of an abstract method, they only force the method to be overridden in the child class. Anyway, nothing prevents us from using base implementation.  The following example demonstrates this behaviour (please don't do it in any production code):

```python
from abc import ABC, abstractmethod


class A(ABC):

    @abstractmethod
    def do_stuff(self):
        return 37


class B(A):

    def do_stuff(self):
        return super().do_stuff() + 2100


if __name__ == '__main__':
    obj = B()
    print(obj.do_stuff())
```

Having said that, I would reword this sentence to better reflect what ABC are about, maybe something like:
```suggestion
The interfaces are located in the ``api`` directory within each packages. They are implemented either as an ``abstract class`` or ``protocols``. `Abstract classes <https://docs.python.org/3/library/abc.html>`_ are classes that contain one or more abstract methods and thus cannot be instantiated. Concrete implementations (subclasses) of a given abstract base class have to override all of the abstract methods. Abstract base classes often serve as blueprint classes, defining common behaviours in terms of some specialized method(s) that has to be implemented in child classes.
```
?